sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs

```diff
@@ -175,12 +174,23 @@ public async Task GetTablesReturnsTablesWithAndWithoutPagination(int? pageCount)
                 }
 
                 // Get the table list.
-
-                var tableResponses = (await service.GetTablesAsync(maxPerPage: pageCount).ToEnumerableAsync().ConfigureAwait(false)).ToList();
-
-                Assert.That(() => tableResponses, Is.Not.Empty);
-                Assert.That(() => tableResponses.Select(r => r.TableName), Contains.Item(tableName));
+                var remainingItems = createdTables.Count;
+                await foreach (var page in service.GetTablesAsync(/*maxPerPage: pageCount*/).AsPages(pageSizeHint: pageCount))
+                {
+                    Assert.That(page.Values, Is.Not.Empty);
+                    if (pageCount.HasValue)
+                    {
+                        Assert.That(page.Values.Count, Is.EqualTo(Math.Min(pageCount.Value, remainingItems)));
```

Comment:
LessThanOrEqualTo?

Reply:
Isn't that less precise than what is there now?

Reply:
True.